### ISSUE TITLE
Doesn't recognize my workspace

### ISSUE THREAD

**Issue by michelmetran** [maikvandergaag/msft-extensions#508]
**Describe the issue**

I created a pipeline to create an artifact, containing the .pbix file.
With the artifact created, I'm trying to publish it to a premium workspace at the company I work for.
(maybe this is the problem:  personal workspaces vs. work workspaces)

However, I am getting the error:
```
2023-10-09T22:08:38.4847323Z ##[debug]Exception:
2023-10-09T22:08:38.4893064Z ##[debug]System.Management.Automation.RuntimeException: Power BI Workspace: CTIC - Indicadores BI - Desenvolvimento does not exist
2023-10-09T22:08:38.5158798Z ##[error]Power BI Workspace: CTIC - Indicadores BI - Desenvolvimento does not exist
2023-10-09T22:08:38.5165302Z ##[debug]Processed: ##vso[task.logissue type=error]Power BI Workspace: CTIC - Indicadores BI - Desenvolvimento does not exist
2023-10-09T22:08:38.5166820Z ##[debug]Processed: ##vso[task.complete result=Failed]
```

<br>

However, the workspace exists!!!

![](https://i.imgur.com/Ijgd0pm.png)

<br>

![](https://i.imgur.com/vp9PL9U.png)

<br>

I noticed the error because I saw, in the logs, that the extension was trying to create the workspace when the creation option was enabled... I thought it was strange, since the workspace in question exists!

When the option was enabled I received an "Invalid Request" message

Then I disabled the option and started getting this error.

![](https://i.imgur.com/HXswIgl.png)

-----

**Extension**
 - Extension: [Power BI Actions](https://marketplace.visualstudio.com/items?itemName=maikvandergaag.maikvandergaag-power-bi-actions)
 - Version: 5.4.44

<br>

-----

**Expected behavior**

Considering that Security Goup is already enabled in the workspace, I expected the dashboard to be published in the workspace.

Note: here is a [video](https://www.youtube.com/watch?v=qskIb2Hilv4) containing explanations for configuring Power BI Actions
Note2: I'm trying to publish to a workspace owned by the company I work for. I have already configured the application and security group registration in the company's Azure Portal!

<br>

-----

**Situation (please complete the following information):**

 - Agent type: windows-2019
 - Autentication: Service Principal

<br>

-----

**Log info**

Attachment [ReleaseLogs_14.zip](https://github.com/maikvandergaag/msft-extensions/files/12850988/ReleaseLogs_14.zip)

**Comment by michelmetran** [comment]
I enabled the option to create the Workspace! But it gives the following error: "InvalidRequest".
I attach the log [ReleaseLogs_16.zip](https://github.com/maikvandergaag/msft-extensions/files/12856792/ReleaseLogs_16.zip)

I also thought the problem was with permissions, but I have ALL permissions enabled for the Service Principal (in section "Power BI Service") and still the error remains.

![](https://i.imgur.com/Z9meuK5.png)

<br>

I had already selected all possible permissions in "Delegated Permissions" and the two possible ones in "Application Permissions", trying to make pbix publishing work...

The only thing I couldn't change (because I don't have permission) is that in the options "Tenant.ReadWrite.All" and "Tenant.Read.All" there is an alert stating that "Not granted for {org}"..

![](https://i.imgur.com/XxLB2sq.png)

<br>

...however, I don't believe this could be related...
I don't know where I'm going wrong...

**Comment by michelmetran** [comment]
@maikvandergaag ,
I appreciate the help!
I've been studying more and more... and I found the error!!
I detail it here to help someone who is also making the error.

I had added Admin permissions to the Group I created in the Azure Portal. It didn't work.
![](https://i.imgur.com/NJm8SNB.png)

I chose to include the Application (Main Service) as Admin! Now it worked! I did this after reading this suggestion in the issue #153
